app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -471,6 +477,14 @@ public void run() {
         }, 20);
     }
 
+    void uploadImmersiveElapsedTime() {
```

Comment:
This function is being called in the render thread. Is TelemetryWrapper.addImmersiveToHistogram() thread safe?

Reply:
yep. It looks risky to put it in the render thread. I have moved it to the main thread.